svm/src/account_loader.rs

```diff
@@ -39,6 +39,41 @@ pub(crate) type TransactionProgramIndices = Vec<Vec<IndexOfAccount>>;
 pub type TransactionCheckResult = Result<CheckedTransactionDetails>;
 pub type TransactionValidationResult = Result<ValidatedTransactionDetails>;
 
+/// This trait abstract the retrieval of transaction check results for a given transaction
+/// in the transaction batch passed to the SVM.
+pub trait TransactionCheckResults {
+    /// The function consumes a check result and returns it, meaning that once it is requested,
+    /// it is no longer available. `idx` is the index of a transaction in the batch.
+    fn consume_check_result(&mut self, _idx: usize) -> TransactionCheckResult {
```

Comment:
it looks like the first thing we do during transaction iteration is call this consume fn.
If that's the case, why not have either an iterator return, or an `IntoIterator` bound?

It seems simple enough to have some dummy struct that impls this with a repeating iterator to get similar behavior, but also doesn't eat (minor) cost of replacement for the vec impl.